homeassistant/components/zwave_js/__init__.py

```diff
@@ -169,6 +135,48 @@ def remove_device(device: device_registry.DeviceEntry) -> None:
         registered_unique_ids.pop(device.id, None)
         discovered_value_ids.pop(device.id, None)
 
+    @callback
+    def register_node_in_dev_reg(node: ZwaveNode) -> device_registry.DeviceEntry:
```

Comment:
I understand why you moved the func back to the async_setup_entry but that is already way to hard to read/understand. I'm leaning towards an intermediate class such as a ZwaveManager where we can have all these methods...

Reply:
My opinion is to leave it outside the async_setup_entry for now as it is (and pass a hell amount of arguments) and we settle on a manager class in a new PR ?

Reply:
Sure, I'll put it back.